### PR TITLE
Add type to session nodes

### PR DESCRIPTION
**Pull request by petermoogk**
## Proposed changes

Currently, the contextValue for a Zowe tree node does not encode the profile type.  This makes it difficult for an extender to add a context menu action that should be visible for a specific profile type.  This PR includes code that updates the context value for each of the three zowe views(ie data set, uss, and jobs).  Each contextValue has "_type=<profile_type>" appended to the end.

## Release Notes

Milestone:

Changelog:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [ ] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [ ] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [ ] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

File: packages/zowe-explorer/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
 - Power users and developers can now build links to efficiently open mainframe resources in Zowe Explorer. Use the **Copy External Link** option in the context menu to get the URL for a data set or USS resource, or create a link in the format `vscode://Zowe.vscode-extension-for-zowe?<ZoweResourceUri>`. For more information on building resource URIs, see the [FileSystemProvider wiki article](https://github.com/zowe/zowe-explorer-vscode/wiki/FileSystemProvider#file-paths-vs-uris). [#3271](https://github.com/zowe/zowe-explorer-vscode/pull/3271)
 - Implemented more user-friendly error messages for API or network errors within Zowe Explorer. [#3243](https://github.com/zowe/zowe-explorer-vscode/pull/3243)
 - Use the "Troubleshoot" option for certain errors to obtain additional context, tips, and resources for how to resolve the errors. [#3243](https://github.com/zowe/zowe-explorer-vscode/pull/3243)
+- The type of a zowe profile is now encoded in the contextValue of a zowe
+tree node.  This allows zowe extenders to add context menu actions to a 
+top level node(ie. ds, uss, or jobs) that is only visible for a particular
+node type.(ie. rse, zosmf, tso, etc)
```
Comment (zFernand0) [inline packages/zowe-explorer/CHANGELOG.md:18]:
```suggestion
- Allowed extenders to add context menu actions to a top level node, i.e. data sets, USS, Jobs, by encoding the profile type in the context value. [#3309](https://github.com/zowe/zowe-explorer-vscode/pull/3309)
```

**Review by zFernand0 — CHANGES_REQUESTED** [review]
Hey @petermoogk,
Curious if we can add (or update) some unit tests that will capture this addition to the context values 🙏

**Review by zFernand0 — CHANGES_REQUESTED** [review]
Hey @petermoogk,

Would you mind adding  (or updating) a small test which verifies that the profile type is added to the context value for each tree?

I'm thinking either a new (or updated) test in each of these files:
- packages/zowe-explorer/__tests__/__unit__/trees/uss/USSTree.unit.test.ts
- packages/zowe-explorer/__tests__/__unit__/trees/dataset/DatasetTree.unit.test.ts
- packages/zowe-explorer/__tests__/__unit__/trees/job/JobTree.unit.test.ts

**Review by traeok — APPROVED** [review]
Changes LGTM, thanks @petermoogk for the enhancement!

**Review by zFernand0 — DISMISSED** [review]
LGTM! 😋

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

**Pull request merged**